pkg/envoy/proxy.go

```diff
@@ -159,14 +160,19 @@ func (e *envoy) args(fname string, overrideFname string) []string {
 	return startupArgs
 }
 
-// readToJSON reads a config file, in YAML or JSON, and returns JSON string
-func readToJSON(fname string) (string, error) {
-	bytes, err := os.ReadFile(fname)
+var HostIP = os.Getenv("HOST_IP")
+
+// readBootstrapToJSON reads a config file, in YAML or JSON, and returns JSON string
+func readBootstrapToJSON(fname string) (string, error) {
+	b, err := os.ReadFile(fname)
 	if err != nil {
 		return "", fmt.Errorf("failed to read file: %s, %v", fname, err)
 	}
 
-	converted, err := yaml.YAMLToJSON(bytes)
+	// Replace host with HOST_IP env var if it is "$(HOST_IP)".
+	// This is to support some tracer setting (Datadog, Zipkin), where "$(HOST_IP)" is used for address.
+	b = bytes.ReplaceAll(b, []byte("$(HOST_IP)"), []byte(HostIP))
```

Comment:
Maybe use shell expansion so it can be easily switched to bash later? ${HOST_IP}?